src/UI/UIView.cpp

```diff
@@ -45,596 +54,905 @@ void UIView::draw_ui_callback()
+          }
+          m_camera_name_window_open = false;
+          ImGui::CloseCurrentPopup();
+        }
+        ImGui::SameLine();
+        if (ImGui::Button("Cancel"))
+        {
+          m_camera_name_window_open = false;
+          m_copy_camera_name.clear();
+          ImGui::CloseCurrentPopup();
+        }
+        ImGui::EndPopup();
+      }
 
-    if (helper::ends_with_string(fileDialog.selected_fn, ".txt"))
-    {
-      VRDataLoader::get_instance()->load_txt_file(m_controller_app, fileDialog.selected_path);
+    }
 
-      // m_controller_app.load_txt_file(fileDialog.GetSelected().string());
+    if (m_camera_animation_duration_open)
+    {
+      ImGui::OpenPopup("Animation Time");
+      ImGui::SetNextWindowSize(ImVec2(350, 400), ImGuiCond_FirstUseEver);
+      if (ImGui::BeginPopupModal("Animation Time", &m_camera_animation_duration_open))
+      {
+        ImGui::Text("Camera Name");
+        ImGui::InputText("##textanimationtime", &m_str_animation_duration, ImGuiInputTextFlags_CharsDecimal);
+        if (ImGui::Button("Ok"))
+        {
+          m_controller_app.get_simulation().set_camera_animation_duration(std::stof(m_str_animation_duration));
+          m_str_animation_duration.clear();
+          m_camera_animation_duration_open = false;
+          ImGui::CloseCurrentPopup();
+        }
+        if (ImGui::Button("Cancel"))
+        {
+          m_camera_animation_duration_open = false;
+          m_str_animation_duration.clear();
+          ImGui::CloseCurrentPopup();
+        }
+        ImGui::EndPopup();
+      }
     }
-#ifdef WITH_TEEM
-    else if (helper::ends_with_string(fileDialog.selected_fn, ".nrrd")) {
-      std::vector<std::string> vals;
-      /*	vals.push_back(fileDialog.GetSelected().string());
-        promises.push_back(new std::promise<Volume*>);
-        futures.push_back(promises.back()->get_future());
-        threads.push_back(new std::thread(&VolumeVisualizationApp::loadVolume, this, vals, promises.back()));*/
+
+
+    //file loading
+    if (m_file_dialog_open)
+    {
+      ImGui::OpenPopup("Open File");
+      m_file_dialog_open = false;
     }
-#endif
-  }
 
 
 
 
-  if (m_file_dialog_save_dir)
-  {
-    ImGui::OpenPopup("Save File");
-    m_file_dialog_save_dir = false;
-    switch (m_current_save_modal)
+    if (fileDialog.showFileDialog("Open File", imgui_addons::ImGuiFileBrowser::DialogMode::OPEN, ImVec2(700, 310), m_file_extension_filter))
     {
-    case SAVE_SESSION:
-      m_save_session_dialog_open = false;
-      break;
-    case SAVE_TRFR_FNC:
-      m_save_trnfct_open = false;
-      break;
-    default:
-      break;
+
+      if (helper::ends_with_string(fileDialog.selected_fn, ".txt"))
+      {
+        //VRDataLoader* insta = VRDataLoader::get_instance();
+        //std::thread t1 ( &VRDataLoader::load_txt_file, std::ref(m_controller_app), fileDialog.selected_path);
+        //t1.join();
```

Comment:
delete commented out code

Reply:
I'll create a new issue to send the load volume routine on a separate thread. This is will help implementing a loading screen in the UI.

Reply:
https://github.com/brown-ccv/VR-Volumeviewer/issues/52